apps/gitness/src/pages-v2/repo/repo-sidebar.tsx

```diff
@@ -45,10 +44,15 @@ export const RepoSidebar = () => {
       git_ref: normalizeGitRef(selectedBranch)
     }
   })
-
-  const branchList: BranchListProps[] | undefined = branches?.body?.map(item => ({
-    name: item?.name || ''
-  }))
+  console.log(branches)
+  const branchList: BranchSelectorListProps[] = useMemo(() => {
+    return (
+      branches?.body?.map(item => ({
+        name: item?.name || '',
+        default: item?.name === repository?.body?.default_branch
+      })) || []
```

Comment:
Nothing wrong here, but you can make it a bit easier to follow by exiting early.
```suggestion
    if (!branches?.body) return []

    return (
      branches.body.map(item => ({
        name: item?.name || '',
        default: item?.name === repository?.body?.default_branch
      }))
```